packages/components/src/Filters/FilterInput.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState, useEffect } from 'react';
+import { Checkbox, Radio } from '@patternfly/react-core';
+
+export interface FilterInputProps {
+  addRemoveFilters?: (value?: string, param?: string, type?: string, checked?: boolean) => void;
+  className?: string;
+  currentPage?: string;
+  id: string;
+  label: React.ReactNode;
+  param: string;
+  value?: string;
+  filters?: Record<string, string>;
+  type?: string;
+  onChange?: (checked: boolean, event: React.FormEvent<HTMLInputElement>) => void;
+}
+
+const FilterInput: React.FC<FilterInputProps> = ({ addRemoveFilters, param, filters, type = 'checkbox', value, id, label }) => {
+  const [checked, setChecked] = useState(false);
+  useEffect(() => {
+    if (type !== 'radio') {
+      if (filters && value && param in filters && filters[param] && filters[param].includes(value)) {
+        setChecked(true);
+      } else {
+        setChecked(false);
+      }
+    }
+  });
```

Comment:
This effect is performed on each render, since we are firing state change in it we'd end in infinite loop. Previously it used to run only on render, which is not ideal as well, but some apps might already operate with it. Let's go for now with just that and later add `filters`, `value` and `param` as dependency of this hook. Also might be better to use safe accessor.
```suggestion
  useEffect(() => {
    if (type !== 'radio') {
      if (value && filters?.[param]?.includes?.(value)) {
        setChecked(true);
      } else {
        setChecked(false);
      }
    }
  }, []);
```